### PR TITLE
Calendar custom code

### PR DESCRIPTION
**Pull request by jlibman**
**NOTE**

I commented out extra functionality (such as viewing the calendar by year or decade), but kept it in the repo in case we'd like to use it.

## Calendar and Event Structure

- Component created with following stucture:
Calendar ->
MonthView ->
Days ->
TileGroup ->
Flex ->
Day ->
Tile -> 
EventCardSlot ->
Event

## Images
screen shot 2018-04-16 at 2 37 50 pm

screen shot 2018-04-15 at 7 41 09 pm

screen shot 2018-04-16 at 11 20 38 am

## TODO

- cite react-calendar
- multiple events in one day

### CODE REVIEW

**Comment by jlibman** [comment]
screen shot 2018-04-16 at 2 44 15 pm

Dialog box css conflict resolved

**Comment by jlibman** [comment]
I haven't looked at safari stuff yet, but I fixed all of the other requested changes, Joyce. However, my z-index doesn't seem to be a problem for the notifications over the calendar..?

screen shot 2018-04-16 at 6 59 03 pm
screen shot 2018-04-16 at 6 58 57 pm

File: src/PageContent/Calendar/Dialog.js [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
 
 class Dialog extends Component {
     render() {
         return (
             <dialog className="event-dialog" open>
                 <Header
-                    eventType={this.props.delivery.eventType}
+                    eventType={this.props.eventType}
                     closeDialog={this.props.closeDialog}
                     date={this.props.delivery.date}
```
Comment (arielin3) [inline src/PageContent/Calendar/Dialog.js]: remove `date`

File: src/PageContent/Calendar/Header.js [hunk 2]
```diff
@@ -29,8 +29,8 @@ class Header extends Component {
                     alt="close"
                 />
                 <p className="header-time">
-                    {this.props.date} {this.props.startTime} -{' '}
-                    {this.props.endTime}
+                    {this.props.startTime.toString()} - +{' '}
+                    {this.props.endTime.toString()}
```
Comment (arielin3) [inline src/PageContent/Calendar/Header.js]: `{moment(this.props.startTimestamp).format(StringFormat.DATE_FULL)}`
`{moment(this.props.startTimestamp).format(StringFormat.TIME)}`
`{moment(this.props.endTimestamp).format(StringFormat.TIME)}`
for each of the strings

File: src/PageContent/Calendar/EventCard/EventCardSlot.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+                let type = DeliveryType.EMERGENCY;
+                if (!this.props.events[e].isEmergency) {
+                    type = DeliveryType.RECURRING;
+                }
+                let startTime = new Date(
+                    this.props.events[e].startTimestamp * 1000
+                );
+                let endTime = new Date(
+                    this.props.events[e].endTimeStamp * 1000
+                );
```
[the remark this answers is not in the document]
Comment (jlibman) [reply src/PageContent/Calendar/EventCard/EventCardSlot.js]: amazing, wouldn't have thought to look in Enums!

File: src/PageContent/Calendar/EventCard/EventCardSlot.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+                // let future = e.startTime
+                //e.startTimestamp to 10am time
+                //e.endTimestamp to 12pm time
+                //futureEvent depends on [after] this.props.today
```
[the remark this answers is not in the document]
Comment (jlibman) [reply src/PageContent/Calendar/EventCard/EventCardSlot.js]: those were just for me - i'll delete them!

File: src/PageContent/Calendar/Header.js (hunk 2 above)
Comment (jlibman) [reply src/PageContent/Calendar/Header.js]: is there a way that has already been defined to incorporate time zones?

screen shot 2018-04-16 at 9 07 04 pm

**Comment by jlibman** [comment]
screen shot 2018-04-19 at 5 35 36 pm
dates updated, code refactored to not iterate over all events for each day

File: src/PageContainer.js [hunk 5]
```diff
@@ -172,27 +65,9 @@ class PageContainer extends Component {
                     handler={this.navBarHandler}
                 />
 
-                {/* TODO: replace placeholder text with real components */}
                 {this.state.content === PageContent.CALENDAR && (
-                    <div style={{ marginTop: '120px', marginLeft: '250px' }}>
+                    <div id="calendar-container">
```
Comment (arielin3) [inline src/PageContainer.js]: nit-pick: move this `div` into `Calendar`

File: src/PageContent/Calendar/ReactCalendar/TileGroup.jsx [hunk 6]
```diff
@@ -0,0 +1,321 @@
...
+            if (date.toString().startsWith(curDay)) {
+                today = true;
+            }
+
+            let dateClass = '';
+            if (today) {
+                dateClass = 'today';
+            }
+
+            for (var deliveryDate in this.state.events) {
```
Comment (arielin3) [inline src/PageContent/Calendar/ReactCalendar/TileGroup.jsx]: no need for the forloop, just need to do:
```
let events = this.state.events[moment(date).format(InputFormat.DATE)];
```

File: src/PageContent/Calendar/EventCard/EventCardSlot.js [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+        this.setState({
+            dialogOpen: false
+        });
+    }
+
+    render() {
+        if (Object.keys(this.props.events).length === 0) {
+            return <div />;
+        } else {
+            for (var e in this.props.events) {
```
Comment (arielin3) [inline src/PageContent/Calendar/EventCard/EventCardSlot.js]: it looks like you do create multiple eventcards for multiple events, so you just need to make them show on as stacked vertically (they're probably on top of each other right now)

File: src/PageContent/Calendar/EventCard/EventCard.js [hunk 8]
```diff
@@ -22,13 +23,15 @@ class EventCard extends Component {
             truckAlt = 'grey truck';
             style = 'event-container event-container-past';
         }
-
+        let startTime = this.props.delivery[0].startTimestamp;
```
Comment (arielin3) [inline src/PageContent/Calendar/EventCard/EventCard.js]: shouldn't the event passed in `this.props.events[e]` be a single `delivery` object? why indexing by 0? same with everywhere else. Each `EventCard` should only deal with 1 delivery

**Comment by jlibman** [comment]
screen shot 2018-04-20 at 8 36 14 pm
**WILL FIX UI FOR MULTIPLE EVENTS IN ONE DAY TO FIT BETTER*

**Comment by arielin3** [comment]
merging to work on backend

**Pull request merged**